spec/sidekiq/evss/disability_compensation_form/form526_document_upload_failure_email_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe EVSS::DisabilityCompensationForm::Form526DocumentUploadFailureEmail, type: :job do
+  let!(:form526_submission) do
+    create(
+      :form526_submission,
+      :with_uploads
+    )
+  end
+
+  let(:upload_data) { [form526_submission.form[Form526Submission::FORM_526_UPLOADS].first] }
+  let(:file) { Rack::Test::UploadedFile.new('spec/fixtures/files/sm_file1.jpg', 'image/jpg') }
+  let!(:form_attachment) do
+    sea = SupportingEvidenceAttachment.new(
+      guid: upload_data.first['confirmationCode']
+    )
+
+    sea.set_file_data!(file)
+    sea.save!
+    sea
+  end
+
+  let(:notification_client) { double('Notifications::Client') }
+
+  before do
+    Sidekiq::Job.clear_all
+    allow(Notifications::Client).to receive(:new).and_return(notification_client)
+  end
+
+  describe '#perform' do
+    let(:formatted_submit_date) do
+      # We display dates in mailers in the format "May 1, 2024 3:01 p.m. EDT"
+      form526_submission.created_at.strftime('%B %-d, %Y %-l:%M %P %Z').sub(/([ap])m/, '\1.m.')
```

Comment:
You should be able to call `format_creation_time_for_mailers` on `form526_submission`.

Reply:
Yeah since I'm testing the correct output format here and effectively testing that method as part of it I wouldn't set an expectation based on what value the method returns. Then it wouldn't really be testing the method at all. As a general rule I try not to set expectations for tests using methods, it obscures the value we are testing for from the person reading the test, even though it would clean up the code